src/talker_character.h

```diff
@@ -0,0 +1,97 @@
+        int num_bionics() const override;
+        bool has_max_power() const override;
+        bool has_bionic( const bionic_id &bionics_id ) const override;
+        std::vector<item *> items_with( const std::function<bool( const item & )> &filter ) const override;
+        bool has_effect( const efftype_id &effect_id ) const override;
+        int get_fatigue() const override;
+        int get_hunger() const override;
+        int get_thirst() const override;
+        tripoint global_omt_location() const override;
+        std::string get_value( const std::string &var_name ) const override;
+        int posx() const override;
+        int posy() const override;
+        int posz() const override;
+        tripoint pos() const override;
+        int cash() const override;
+        bool knows_spell( const spell_id &sp ) const override;
+        bool unarmed_attack() const override;
+        bool in_control( const vehicle &veh ) const override;
+        bool can_stash_weapon() const override;
+        bool has_stolen_item( const talker &guy ) const override;
+        int get_skill_level( const skill_id &skill ) const override;
+        bool is_male() const override;
+        // override functions called in npctalk.cpp
+        std::vector<std::string> get_topics( const bool ) override;
+        bool is_deaf() const override;
+        std::string short_description() const override;
+        std::vector<std::string> get_grammatical_genders() const override;
+        character_id getID() const override;
+        faction *get_faction() const override;
+        int parse_mod( const std::string &attribute, const int factor ) const override;
+        int trial_chance_mod( const std::string &trial_type ) const override;
+        // override functions called in npctalk.cpp
+        void shout( const std::string &speech = "", bool order = false ) override;
+        void add_effect( const efftype_id &new_effect, const time_duration &dur,
+                         const bool permanent ) override;
+        void remove_effect( const efftype_id &old_effect ) override;
+        void set_mutation( const trait_id &new_trait ) override;
+        void unset_mutation( const trait_id &old_trait ) override;
+        void set_value( const std::string &var_name, const std::string &value ) override;
+        void remove_value( const std::string &var_name ) override;
+        void i_add( const item &new_item ) override;
+        std::list<item> use_charges( const itype_id &item_name, const int count ) override;
+        std::list<item> use_amount( const itype_id &item_name, const int count ) override;
+        void remove_items_with( const std::function<bool( const item & )> &filter ) override;
+        void buy_monster( talker &seller, const mtype_id &mtype, int cost,
+                          int count, bool pacified, const translation &name ) override;
+    protected:
+        mutable player *me_chr;
```

Comment:
Why is this `mutable`?  That seems like a bad sign...

Reply:
there were a couple of places where a player or npc member function was not const but had to be called from a const function for some other reason.  I got tired of fighting with the compiler and went for the drastic solution, but I was pretty fed up with the compiler at this point.  Let me retry and see if I can work out the logic safely.

Reply:
mutable removed, thanks for reminder.